### PR TITLE
Fixed issue related with previous change in QueryDimInterface.

### PR DESCRIPTION
**Pull request by bastiao**
Fixed issue related with previous change in QueryDimInterface. The query interface was not able to reach the correct interfaces at plugins.

### CODE REVIEW

File: dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java [hunk 1]
```diff
@@ -603,8 +622,35 @@ public JointQueryTask query(JointQueryTask holder, List<String> querySources, fi
...
+        //returns a tasks that runs the query from the selected query engine
+        String uid = UUID.randomUUID().toString();
+        Task<Iterable<SearchResult>> queryTask = new Task<>(uid, querySource,
+                new Callable<Iterable<SearchResult>>(){
+                    @Override public Iterable<SearchResult> call() throws Exception {
+                        if(queryEngine == null || !(queryEngine instanceof QueryDimInterface)) return Collections.emptyList();
+                        try {
+                            return queryEngine.query(query, level, parameters);
+                        } catch (RuntimeException ex) {
+                            logger.warn("Query plugin {} failed unexpectedly", querySource, ex);
```
Comment (Enet4) [inline dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java]: A previously unattended issue: should public `query` methods in `DicooglePlatform` swallow the query-related exceptions `QueryException` and `RetrievalException`? Can we make them appear when attempting to get the outcome of the task?

File: dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java [hunk 2]
```diff
@@ -483,7 +483,7 @@ public QueryInterface getQueryProviderByName(String name, boolean onlyEnabled){
         List<String> baseProviders = ServerSettingsManager.getSettings()
                 .getArchiveSettings().getDIMProviders();
 
-        if (baseProviders.isEmpty()) {
+        if (baseProviders==null || baseProviders.isEmpty()) {
```
Comment (Enet4) [inline dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java]: Good catch. I had the feeling that this wasn't working properly in my recent uses of Dicoogle 3.

File: dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java [hunk 3]
```diff
@@ -603,8 +622,35 @@ public JointQueryTask query(JointQueryTask holder, List<String> querySources, fi
...
+    private Task<Iterable<SearchResult>> getTaskForQueryDim(final String querySource, final String query,
+                                                         final DimLevel level, final Object ... parameters){
+
+        final QueryInterface queryEngine = getQueryProviderByName(querySource, true);
+        //returns a tasks that runs the query from the selected query engine
+        String uid = UUID.randomUUID().toString();
+        Task<Iterable<SearchResult>> queryTask = new Task<>(uid, querySource,
+                new Callable<Iterable<SearchResult>>(){
+                    @Override public Iterable<SearchResult> call() throws Exception {
+                        if(queryEngine == null || !(queryEngine instanceof QueryDimInterface)) return Collections.emptyList();
```
Comment (Enet4) [inline dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java]: A more fail-fast approach would be performing this check before the task is created, which should be fine if you're OK with throwing when the program fails to create the task.

File: dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java (hunk 1 above)
Comment (bastiao) [reply dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java]: LGTM!

File: dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java (hunk 2 above)
Comment (bastiao) [reply dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java]: Yeap!

File: dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java (hunk 3 above)
Comment (bastiao) [reply dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java]: This is a private method - safer and comfortable to return an empty list. No clear benefit to throw exception IMHO and throw error at this stage.

**Pull request merged**